File: Section-4-Injections/example1_vulnerable_code.py
import sqlite3

class SocialMedia:
    def get_profile_by_id(self, user_id):
        connection = sqlite3.connect("social_media.db")
        cursor = connection.cursor()

        # Vulnerable to SQL Injection
        query = f"SELECT * FROM users WHERE user_id = '{user_id}'"
        cursor.execute(query)
        result = cursor.fetchone()

        connection.close()

        if result:
            return f"Name: {result[1]}, Email: {result[2]}"
        else:
            return "User not found."

# Example usage
social_media = SocialMedia()
print(social_media.get_profile_by_id("1"))

File: Section-4-Injections/example1_vulnerable_code.java
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.ResultSet;
import java.sql.Statement;

public class CustomerProfile {
    public String getCustomerProfileById(String customerId) {
        String query = "SELECT * FROM Customers WHERE CustomerId = '" + customerId + "'";

        try (Connection connection = DriverManager.getConnection(connectionString, "username", "password");
             Statement statement = connection.createStatement();
             ResultSet resultSet = statement.executeQuery(query)) {

            if (resultSet.next()) {
                return "Name: " + resultSet.getString("Name") + ", Email: " + resultSet.getString("Email");
            } else {
                return "Customer not found.";
            }
        } catch (Exception e) {
            e.printStackTrace();
            return "Error retrieving customer profile.";
        }
    }

    public static void main(String[] args) {
        CustomerProfile profile = new CustomerProfile();
        System.out.println(profile.getCustomerProfileById("123"));
    }
}

